txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +68,20 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            Uid outboundMappingUid = new Uid();
```

Comment:
How about:
private static final ConcurrentMap<Transaction, org.jboss.jbossts.txbridge.outbound.OutboundBridge> transctionToUidMappings = new ConcurrentHashMap<>();

That was not guaranteed to work with JTA < 1.2 but as per https://issues.jboss.org/browse/JBTM-2848 it should be safe now